In [2]:
#import dataset
import pandas as pd
import os
customers_df = pd.read_csv(r"C:\Users\THIRU\OneDrive\Desktop\data_illuminate\customers.csv",encoding='latin1')

In [3]:
customers_df.head()

CustomerKey  Gender               Name            City State Code  \
0          301  Female      Lilly Harding  WANDEARAH EAST         SA   
1          325  Female       Madison Hull      MOUNT BUDD         WA   
2          554  Female      Claire Ferres       WINJALLOK        VIC   
3          786    Male  Jai Poltpalingada    MIDDLE RIVER         SA   
4         1042    Male    Aidan Pankhurst   TAWONGA SOUTH        VIC   

               State Zip Code    Country  Continent    Birthday  
0    South Australia     5523  Australia  Australia    7/3/1939  
1  Western Australia     6522  Australia  Australia   9/27/1979  
2           Victoria     3380  Australia  Australia   5/26/1947  
3    South Australia     5223  Australia  Australia   9/17/1957  
4           Victoria     3698  Australia  Australia  11/19/1965

standardize strings

In [4]:
customers_df['Gender'] = customers_df['Gender'].str.strip().str.capitalize()
customers_df['Name'] = customers_df['Name'].str.strip().str.title()
customers_df['City'] = customers_df['City'].str.strip().str.title()
customers_df['State'] = customers_df['State'].str.strip().str.title()
customers_df['State Code'] = customers_df['State Code'].str.strip().str.upper()
customers_df['Country'] = customers_df['Country'].str.strip().str.title()
customers_df['Continent'] = customers_df['Continent'].str.strip().str.title()

check for null values

In [5]:
customers_df.isnull().sum()

CustomerKey     0
Gender          0
Name            0
City            0
State Code     10
State           0
Zip Code        0
Country         0
Continent       0
Birthday        0
dtype: int64

In [6]:
customers_df[customers_df['State Code'].isnull()]

CustomerKey  Gender                Name               City State Code  \
5304       729681  Female    Rossana Padovesi            Polvica        NaN   
5316       732289    Male        Indro Piccio          Varcaturo        NaN   
5372       742042    Male     Amaranto Loggia          Casaferro        NaN   
5377       742886  Female       Edmonda Capon           Terzigno        NaN   
5378       743343  Female        Ambra Sagese  Pomigliano D'Arco        NaN   
5485       759705    Male    Callisto Lo Duca            Casilli        NaN   
5525       765589    Male  Michelino Lucchesi       Pompei Scavi        NaN   
5531       766410    Male   Adelmio Beneventi             Licola        NaN   
5631       781667  Female          Ilda Manna             Napoli        NaN   
5695       789177    Male   Calogero Folliero      Mariglianella        NaN   

       State Zip Code Country Continent    Birthday  
5304  Napoli    80035   Italy    Europe   4/18/1981  
5316  Napoli    80014   Italy    Europe   2/24/1949  
5372  Napoli    80034   Italy    Europe   3/14/1936  
5377  Napoli    80040   Italy    Europe    8/6/1963  
5378  Napoli    80038   Italy    Europe    1/5/1961  
5485  Napoli    80047   Italy    Europe   8/28/1976  
5525  Napoli    80045   Italy    Europe  11/13/1947  
5531  Napoli    80078   Italy    Europe   1/13/1940  
5631  Napoli    80134   Italy    Europe    5/8/1977  
5695  Napoli    80030   Italy    Europe    3/3/2000

fill null values of state code with corresponding country values

In [7]:
customers_df.loc[(customers_df['Country']=='Italy') & (customers_df['State Code'].isnull()),'State Code'] = 'IT'

Convert Birthday column to datetime format

In [8]:
customers_df['Birthday'] = pd.to_datetime(customers_df['Birthday'],format='%m/%d/%Y')

In [9]:
customers_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15266 entries, 0 to 15265
Data columns (total 10 columns):
 #   Column       Non-Null Count  Dtype         
---  ------       --------------  -----         
 0   CustomerKey  15266 non-null  int64         
 1   Gender       15266 non-null  object        
 2   Name         15266 non-null  object        
 3   City         15266 non-null  object        
 4   State Code   15266 non-null  object        
 5   State        15266 non-null  object        
 6   Zip Code     15266 non-null  object        
 7   Country      15266 non-null  object        
 8   Continent    15266 non-null  object        
 9   Birthday     15266 non-null  datetime64[ns]
dtypes: datetime64[ns](1), int64(1), object(8)
memory usage: 1.2+ MB


Check for duplicate records

In [10]:
customers_df.duplicated().sum()

np.int64(0)

Rename columns

In [11]:
customers_df.columns

Index(['CustomerKey', 'Gender', 'Name', 'City', 'State Code', 'State',
       'Zip Code', 'Country', 'Continent', 'Birthday'],
      dtype='object')

In [12]:
customer_df = customers_df.rename(columns={
    'CustomerKey': 'customer_key',
    'Gender': 'gender',
    'City': 'city',
    'State Code': 'state_code',
    'State': 'state',
    'Zip Code': 'zip_code',
    'Country': 'country',
    'Continent': 'continent',
    'Birthday': 'birthday'
})


Export cleaned dataset

In [77]:
# Define the directory and file path
directory = 'cleaned_dataset'
file_path = os.path.join(directory, 'products.csv')

# Create the directory if it doesn't exist
if not os.path.exists(directory):
    os.makedirs(directory)

# Save the DataFrame to the specified file path
customers_df.to_csv(file_path, index=False)
print("File saved successfully.")

File saved successfully.


check dataframe

In [15]:
customers_df

CustomerKey  Gender               Name                 City State Code  \
0              301  Female      Lilly Harding       Wandearah East         SA   
1              325  Female       Madison Hull           Mount Budd         WA   
2              554  Female      Claire Ferres            Winjallok        VIC   
3              786    Male  Jai Poltpalingada         Middle River         SA   
4             1042    Male    Aidan Pankhurst        Tawonga South        VIC   
...            ...     ...                ...                  ...        ...   
15261      2099600  Female     Denisa DušKová              Houston         TX   
15262      2099618    Male   Justin Solórzano               Mclean         VA   
15263      2099758    Male    Svend Petrussen           Wilmington         NC   
15264      2099862  Female       Lorenza Rush            Riverside         CA   
15265      2099937    Male   Zygmunt Kaminski  Bloomfield Township         MI   

                   State Zip Code        Country      Continent   Birthday  
0        South Australia     5523      Australia      Australia 1939-07-03  
1      Western Australia     6522      Australia      Australia 1979-09-27  
2               Victoria     3380      Australia      Australia 1947-05-26  
3        South Australia     5223      Australia      Australia 1957-09-17  
4               Victoria     3698      Australia      Australia 1965-11-19  
...                  ...      ...            ...            ...        ...  
15261              Texas    77017  United States  North America 1936-03-25  
15262           Virginia    22101  United States  North America 1992-02-16  
15263     North Carolina    28405  United States  North America 1937-11-09  
15264         California    92501  United States  North America 1937-10-12  
15265           Michigan    48302  United States  North America 1965-08-18  

[15266 rows x 10 columns]

# products

In [18]:
products_df = pd.read_csv(r'C:\Users\THIRU\OneDrive\Desktop\data_illuminate\products.csv')

In [19]:
products_df

ProductKey                                   Product Name    Brand  \
0              1            Contoso 512MB MP3 Player E51 Silver  Contoso   
1              2              Contoso 512MB MP3 Player E51 Blue  Contoso   
2              3               Contoso 1G MP3 Player E100 White  Contoso   
3              4              Contoso 2G MP3 Player E200 Silver  Contoso   
4              5                 Contoso 2G MP3 Player E200 Red  Contoso   
...          ...                                            ...      ...   
2512        2513    Contoso Bluetooth Active Headphones L15 Red  Contoso   
2513        2514  Contoso Bluetooth Active Headphones L15 White  Contoso   
2514        2515             Contoso In-Line Coupler E180 White  Contoso   
2515        2516             Contoso In-Line Coupler E180 Black  Contoso   
2516        2517            Contoso In-Line Coupler E180 Silver  Contoso   

       Color Unit Cost USD Unit Price USD  SubcategoryKey  \
0     Silver        $6.62         $12.99              101   
1       Blue        $6.62         $12.99              101   
2      White        $7.40         $14.52              101   
3     Silver       $11.00         $21.57              101   
4        Red       $11.00         $21.57              101   
...      ...           ...            ...             ...   
2512     Red       $43.07        $129.99              505   
2513   White       $43.07        $129.99              505   
2514   White        $1.71          $3.35              505   
2515   Black        $1.71          $3.35              505   
2516  Silver        $1.71          $3.35              505   

                  Subcategory  CategoryKey     Category  
0                     MP4&MP3            1        Audio  
1                     MP4&MP3            1        Audio  
2                     MP4&MP3            1        Audio  
3                     MP4&MP3            1        Audio  
4                     MP4&MP3            1        Audio  
...                       ...          ...          ...  
2512  Cell phones Accessories            5  Cell phones  
2513  Cell phones Accessories            5  Cell phones  
2514  Cell phones Accessories            5  Cell phones  
2515  Cell phones Accessories            5  Cell phones  
2516  Cell phones Accessories            5  Cell phones  

[2517 rows x 10 columns]

In [20]:
products_df['Product Name'] = products_df['Product Name'].str.strip()
products_df['Brand'] = products_df['Brand'].str.strip()
products_df['Color'] = products_df['Color'].str.strip()
products_df['Subcategory'] = products_df['Subcategory'].str.strip()
products_df['Category'] = products_df['Category'].str.strip()

Check for null or missing values

In [21]:
products_df.isna().sum()

ProductKey        0
Product Name      0
Brand             0
Color             0
Unit Cost USD     0
Unit Price USD    0
SubcategoryKey    0
Subcategory       0
CategoryKey       0
Category          0
dtype: int64

check for duplicate rows

In [22]:
products_df.duplicated().sum()

np.int64(0)

check for datatypes

In [23]:
products_df.dtypes

ProductKey         int64
Product Name      object
Brand             object
Color             object
Unit Cost USD     object
Unit Price USD    object
SubcategoryKey     int64
Subcategory       object
CategoryKey        int64
Category          object
dtype: object

Extract int values from price column and reassign it with float as dtype

In [24]:
# First convert to float, then format
products_df['Unit Cost USD'] = products_df['Unit Cost USD'].str.extract(r'(\d+\.\d+|\d+)')[0].astype(float).round(2)
products_df['Unit Price USD'] = products_df['Unit Price USD'].str.extract(r'(\d+\.\d+|\d+)')[0].astype(float).round(2)

In [25]:
products_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2517 entries, 0 to 2516
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   ProductKey      2517 non-null   int64  
 1   Product Name    2517 non-null   object 
 2   Brand           2517 non-null   object 
 3   Color           2517 non-null   object 
 4   Unit Cost USD   2517 non-null   float64
 5   Unit Price USD  2517 non-null   float64
 6   SubcategoryKey  2517 non-null   int64  
 7   Subcategory     2517 non-null   object 
 8   CategoryKey     2517 non-null   int64  
 9   Category        2517 non-null   object 
dtypes: float64(2), int64(3), object(5)
memory usage: 196.8+ KB


check stats

In [26]:
products_df.describe()

ProductKey  Unit Cost USD  Unit Price USD  SubcategoryKey  CategoryKey
count  2517.000000    2517.000000     2517.000000     2517.000000  2517.000000
mean   1259.000000     141.763993      235.676432      491.810091     4.878824
std     726.739637     171.594673      239.725973      229.887134     2.299170
min       1.000000       0.480000        0.950000      101.000000     1.000000
25%     630.000000      30.580000       39.990000      305.000000     3.000000
50%    1259.000000      86.140000      165.000000      406.000000     4.000000
75%    1888.000000     183.540000      326.000000      801.000000     8.000000
max    2517.000000     960.820000      999.900000      808.000000     8.000000

Rename columns

In [27]:
products_df.columns

Index(['ProductKey', 'Product Name', 'Brand', 'Color', 'Unit Cost USD',
       'Unit Price USD', 'SubcategoryKey', 'Subcategory', 'CategoryKey',
       'Category'],
      dtype='object')

In [28]:
products_df = products_df.rename(columns={
    'ProductKey': 'product_key',
    'Product Name': 'product_name',
    'Brand': 'brand',
    'Color': 'color',
    'Unit Cost USD': 'unit_cost_usd',
    'Unit Price USD': 'unit_price_usd',
    'SubcategoryKey': 'sub_category_key',
    'Subcategory': 'sub_category',
    'CategoryKey': 'category_key',
    'Category': 'category'
    })

Export cleaned Dataset

In [29]:
# Define the directory and file path
directory = 'cleaned_dataset'
file_path = os.path.join(directory, 'products.csv')

# Create the directory if it doesn't exist
if not os.path.exists(directory):
    os.makedirs(directory)

# Save the DataFrame to the specified file path
products_df.to_csv(file_path, index=False)
print("File saved successfully.")

File saved successfully.


check dataset

In [30]:
products_df

product_key                                   product_name    brand  \
0               1            Contoso 512MB MP3 Player E51 Silver  Contoso   
1               2              Contoso 512MB MP3 Player E51 Blue  Contoso   
2               3               Contoso 1G MP3 Player E100 White  Contoso   
3               4              Contoso 2G MP3 Player E200 Silver  Contoso   
4               5                 Contoso 2G MP3 Player E200 Red  Contoso   
...           ...                                            ...      ...   
2512         2513    Contoso Bluetooth Active Headphones L15 Red  Contoso   
2513         2514  Contoso Bluetooth Active Headphones L15 White  Contoso   
2514         2515             Contoso In-Line Coupler E180 White  Contoso   
2515         2516             Contoso In-Line Coupler E180 Black  Contoso   
2516         2517            Contoso In-Line Coupler E180 Silver  Contoso   

       color  unit_cost_usd  unit_price_usd  sub_category_key  \
0     Silver           6.62           12.99               101   
1       Blue           6.62           12.99               101   
2      White           7.40           14.52               101   
3     Silver          11.00           21.57               101   
4        Red          11.00           21.57               101   
...      ...            ...             ...               ...   
2512     Red          43.07          129.99               505   
2513   White          43.07          129.99               505   
2514   White           1.71            3.35               505   
2515   Black           1.71            3.35               505   
2516  Silver           1.71            3.35               505   

                 sub_category  category_key     category  
0                     MP4&MP3             1        Audio  
1                     MP4&MP3             1        Audio  
2                     MP4&MP3             1        Audio  
3                     MP4&MP3             1        Audio  
4                     MP4&MP3             1        Audio  
...                       ...           ...          ...  
2512  Cell phones Accessories             5  Cell phones  
2513  Cell phones Accessories             5  Cell phones  
2514  Cell phones Accessories             5  Cell phones  
2515  Cell phones Accessories             5  Cell phones  
2516  Cell phones Accessories             5  Cell phones  

[2517 rows x 10 columns]

# sales

In [32]:
sales_df = pd.read_csv(r'C:\Users\THIRU\OneDrive\Desktop\data_illuminate\sales.csv')

In [33]:
sales_df.head(3)

Order Number  Line Item Order Date Delivery Date  CustomerKey  StoreKey  \
0        366000          1   1/1/2016           NaN       265598        10   
1        366001          1   1/1/2016     1/13/2016      1269051         0   
2        366001          2   1/1/2016     1/13/2016      1269051         0   

   ProductKey  Quantity Currency Code  
0        1304         1           CAD  
1        1048         2           USD  
2        2007         1           USD

check for null values

In [34]:
sales_df.isnull().sum()

Order Number         0
Line Item            0
Order Date           0
Delivery Date    49719
CustomerKey          0
StoreKey             0
ProductKey           0
Quantity             0
Currency Code        0
dtype: int64

drop delivery date column as it contains 49719 null records

In [35]:
sales_df = sales_df.drop(columns=['Delivery Date'])

check for duplicates

In [36]:
sales_df.duplicated().sum()

np.int64(0)

data type conversion

In [38]:
sales_df.dtypes

Order Number      int64
Line Item         int64
Order Date       object
CustomerKey       int64
StoreKey          int64
ProductKey        int64
Quantity          int64
Currency Code    object
dtype: object

In [39]:
sales_df['Order Date'] = pd.to_datetime(sales_df['Order Date'],format='%m/%d/%Y')

standardize text data

In [40]:
sales_df['Currency Code'] = sales_df['Currency Code'].str.strip().str.upper()

Rename columns

In [41]:
sales_df.columns

Index(['Order Number', 'Line Item', 'Order Date', 'CustomerKey', 'StoreKey',
       'ProductKey', 'Quantity', 'Currency Code'],
      dtype='object')

In [42]:
sales_df = sales_df.rename(columns={
    'Order Number': 'order_number',
    'Line Item': 'line_item',
    'Order Date': 'order_date',
    'CustomerKey': 'customer_key',
    'StoreKey': 'store_key',
    'ProductKey': 'product_key',
    'Quantity': 'quantity',
    'Currency Code': 'currency_code',
})

In [43]:
sales_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 62884 entries, 0 to 62883
Data columns (total 8 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   order_number   62884 non-null  int64         
 1   line_item      62884 non-null  int64         
 2   order_date     62884 non-null  datetime64[ns]
 3   customer_key   62884 non-null  int64         
 4   store_key      62884 non-null  int64         
 5   product_key    62884 non-null  int64         
 6   quantity       62884 non-null  int64         
 7   currency_code  62884 non-null  object        
dtypes: datetime64[ns](1), int64(6), object(1)
memory usage: 3.8+ MB


In [44]:
sales_df.describe()

order_number     line_item                     order_date  \
count  6.288400e+04  62884.000000                          62884   
mean   1.430905e+06      2.164207  2018-11-30 21:21:56.939126016   
min    3.660000e+05      1.000000            2016-01-01 00:00:00   
25%    1.121017e+06      1.000000            2018-01-25 00:00:00   
50%    1.498016e+06      2.000000            2019-02-06 00:00:00   
75%    1.788010e+06      3.000000            2019-11-23 00:00:00   
max    2.243032e+06      7.000000            2021-02-20 00:00:00   
std    4.532963e+05      1.365170                            NaN   

       customer_key     store_key   product_key      quantity  
count  6.288400e+04  62884.000000  62884.000000  62884.000000  
mean   1.180797e+06     31.802144   1125.859344      3.144790  
min    3.010000e+02      0.000000      1.000000      1.000000  
25%    6.808580e+05      8.000000    437.000000      1.000000  
50%    1.261200e+06     37.000000   1358.000000      2.000000  
75%    1.686496e+06     53.000000   1650.000000      4.000000  
max    2.099937e+06     66.000000   2517.000000     10.000000  
std    5.859634e+05     22.978188    709.244010      2.256371

Export cleaned dataset

In [45]:
sales_df.to_csv('cleaned_dataset/sales.csv',index=False)

In [46]:
sales_df

order_number  line_item order_date  customer_key  store_key  \
0            366000          1 2016-01-01        265598         10   
1            366001          1 2016-01-01       1269051          0   
2            366001          2 2016-01-01       1269051          0   
3            366002          1 2016-01-01        266019          0   
4            366002          2 2016-01-01        266019          0   
...             ...        ...        ...           ...        ...   
62879       2243030          1 2021-02-20       1216913         43   
62880       2243031          1 2021-02-20        511229          0   
62881       2243032          1 2021-02-20        331277          0   
62882       2243032          2 2021-02-20        331277          0   
62883       2243032          3 2021-02-20        331277          0   

       product_key  quantity currency_code  
0             1304         1           CAD  
1             1048         2           USD  
2             2007         1           USD  
3             1106         7           CAD  
4              373         1           CAD  
...            ...       ...           ...  
62879          632         3           USD  
62880           98         4           EUR  
62881         1613         2           CAD  
62882         1717         2           CAD  
62883          464         7           CAD  

[62884 rows x 8 columns]

# store

In [47]:
stores_df = pd.read_csv(r'C:\Users\THIRU\OneDrive\Desktop\data_illuminate\stores.csv')

In [48]:
stores_df.head()

StoreKey    Country                         State  Square Meters  Open Date
0         1  Australia  Australian Capital Territory          595.0   1/1/2008
1         2  Australia            Northern Territory          665.0  1/12/2008
2         3  Australia               South Australia         2000.0   1/7/2012
3         4  Australia                      Tasmania         2000.0   1/1/2010
4         5  Australia                      Victoria         2000.0  12/9/2015

In [49]:
stores_df.info() 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 67 entries, 0 to 66
Data columns (total 5 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   StoreKey       67 non-null     int64  
 1   Country        67 non-null     object 
 2   State          67 non-null     object 
 3   Square Meters  66 non-null     float64
 4   Open Date      67 non-null     object 
dtypes: float64(1), int64(1), object(3)
memory usage: 2.7+ KB


Open Date column has wrong datatype instead of datetime it has object so, altering it

In [50]:
stores_df['Open Date'] = pd.to_datetime(stores_df['Open Date'],format='%m/%d/%Y')

In [51]:

stores_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 67 entries, 0 to 66
Data columns (total 5 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   StoreKey       67 non-null     int64         
 1   Country        67 non-null     object        
 2   State          67 non-null     object        
 3   Square Meters  66 non-null     float64       
 4   Open Date      67 non-null     datetime64[ns]
dtypes: datetime64[ns](1), float64(1), int64(1), object(2)
memory usage: 2.7+ KB


In [52]:
stores_df.isna().sum()

StoreKey         0
Country          0
State            0
Square Meters    1
Open Date        0
dtype: int64

In [53]:
stores_df.describe()

StoreKey  Square Meters                      Open Date
count  67.000000      66.000000                             67
mean   33.000000    1402.196970  2011-06-24 00:42:59.104477696
min     0.000000     245.000000            2005-03-04 00:00:00
25%    16.500000    1108.750000            2009-06-03 00:00:00
50%    33.000000    1347.500000            2010-06-03 00:00:00
75%    49.500000    2000.000000            2013-06-07 00:00:00
max    66.000000    2105.000000            2019-03-05 00:00:00
std    19.485037     576.404058                            NaN

In [54]:
stores_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 67 entries, 0 to 66
Data columns (total 5 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   StoreKey       67 non-null     int64         
 1   Country        67 non-null     object        
 2   State          67 non-null     object        
 3   Square Meters  66 non-null     float64       
 4   Open Date      67 non-null     datetime64[ns]
dtypes: datetime64[ns](1), float64(1), int64(1), object(2)
memory usage: 2.7+ KB


Check Duplicates

In [55]:
stores_df.duplicated().sum() 

np.int64(0)

Standardize Text Columns

In [56]:
stores_df['Country'] = stores_df['Country'].str.strip().str.lower()
stores_df['State'] = stores_df['State'].str.strip().str.lower()

In [57]:
stores_df.head()

StoreKey    Country                         State  Square Meters  Open Date
0         1  australia  australian capital territory          595.0 2008-01-01
1         2  australia            northern territory          665.0 2008-01-12
2         3  australia               south australia         2000.0 2012-01-07
3         4  australia                      tasmania         2000.0 2010-01-01
4         5  australia                      victoria         2000.0 2015-12-09

Check for Invalid Values (check if any unwanted text or format)

In [58]:
stores_df['Country'].unique()

array(['australia', 'canada', 'france', 'germany', 'italy', 'netherlands',
       'united kingdom', 'united states', 'online'], dtype=object)

In [59]:
stores_df['State'].unique()

array(['australian capital territory', 'northern territory',
       'south australia', 'tasmania', 'victoria', 'western australia',
       'new brunswick', 'newfoundland and labrador',
       'northwest territories', 'nunavut', 'yukon', 'basse-normandie',
       'corse', 'franche-comté', 'la réunion', 'limousin', 'martinique',
       'mayotte', 'berlin', 'brandenburg', 'freie hansestadt bremen',
       'freistaat thüringen', 'hamburg', 'hessen',
       'mecklenburg-vorpommern', 'saarland', 'sachsen-anhalt',
       'caltanissetta', 'enna', 'pesaro', 'drenthe', 'flevoland',
       'friesland', 'groningen', 'zeeland', 'armagh', 'ayrshire',
       'belfast', 'blaenau gwent', 'dungannon and south tyrone',
       'fermanagh', 'north down', 'alaska', 'arkansas', 'connecticut',
       'delaware', 'hawaii', 'idaho', 'iowa', 'kansas', 'maine',
       'mississippi', 'montana', 'nebraska', 'nevada', 'new hampshire',
       'new mexico', 'north dakota', 'oregon', 'rhode island',
       'south carol

Rename columns

In [60]:
stores_df.columns

Index(['StoreKey', 'Country', 'State', 'Square Meters', 'Open Date'], dtype='object')

In [61]:
store_df = stores_df.rename(columns={
    'StoreKey': 'store_key',
    'Country': 'country',
    'State': 'state',
    'Square Meters': 'square_meters',
    'Open Date': 'open_date'
})

Export Cleaned Data

In [62]:
stores_df.to_csv('cleaned_dataset/stores.csv',index=False)

In [63]:
stores_df

StoreKey        Country                         State  Square Meters  \
0          1      australia  australian capital territory          595.0   
1          2      australia            northern territory          665.0   
2          3      australia               south australia         2000.0   
3          4      australia                      tasmania         2000.0   
4          5      australia                      victoria         2000.0   
..       ...            ...                           ...            ...   
62        63  united states                          utah         2000.0   
63        64  united states                 washington dc         1330.0   
64        65  united states                 west virginia         1785.0   
65        66  united states                       wyoming          840.0   
66         0         online                        online            NaN   

    Open Date  
0  2008-01-01  
1  2008-01-12  
2  2012-01-07  
3  2010-01-01  
4  2015-12-09  
..        ...  
62 2008-03-06  
63 2010-01-01  
64 2012-01-01  
65 2014-01-01  
66 2010-01-01  

[67 rows x 5 columns]

# exchange rates


In [66]:
rates_df = pd.read_csv(r'C:\Users\THIRU\OneDrive\Desktop\data_illuminate\rates.csv')

In [67]:
rates_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11215 entries, 0 to 11214
Data columns (total 3 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Date      11215 non-null  object 
 1   Currency  11215 non-null  object 
 2   Exchange  11215 non-null  float64
dtypes: float64(1), object(2)
memory usage: 263.0+ KB


Convert date column type from object to datetime

In [68]:
rates_df['Date'] = pd.to_datetime(rates_df['Date'],format='%m/%d/%Y')

In [69]:
rates_df['Currency'].unique()

array(['USD', 'CAD', 'AUD', 'EUR', 'GBP'], dtype=object)

remove any unwanted spaces and ensure all in uppercase

In [70]:
rates_df['Currency'] = rates_df['Currency'].str.strip().str.upper()

check for duplicate entries

In [71]:
rates_df.duplicated().sum()

np.int64(0)

In [73]:
rates_df.columns

Index(['Date', 'Currency', 'Exchange'], dtype='object')

In [74]:
rates_df = rates_df.rename(columns={
    'Date': 'date',
    'Currency': 'currency',
    'Exchange': 'exchange'
    })

In [75]:
rates_df.to_csv('cleaned_dataset/rates.csv',index=False)

In [76]:
rates_df 

date currency  exchange
0     2015-01-01      USD    1.0000
1     2015-01-01      CAD    1.1583
2     2015-01-01      AUD    1.2214
3     2015-01-01      EUR    0.8237
4     2015-01-01      GBP    0.6415
...          ...      ...       ...
11210 2021-02-20      USD    1.0000
11211 2021-02-20      CAD    1.2610
11212 2021-02-20      AUD    1.2723
11213 2021-02-20      EUR    0.8238
11214 2021-02-20      GBP    0.7126

[11215 rows x 3 columns]